src/scripts/ci_check_install.py

```diff
@@ -12,44 +12,81 @@
 
 import os
 import sys
+import json
+import re
 
-def has_extension(filename, extensions):
-    for ext in extensions:
-        if filename.endswith(".%s" % ext):
-            return True
-    return False
+def verify_library(build_config):
+    lib_dir = build_config['libdir']
+    if not os.path.isdir(lib_dir):
+        print('Error: libdir "%s" is not a directory' % lib_dir)
+        return False
 
-def is_lib_file(filename):
-    return has_extension(filename, ["so", "a", "dll", "dylib", "lib"])
+    found_libs = set([])
 
-def is_header_file(filename):
-    return has_extension(filename, ["h", "hpp", "h++", "hxx", "hh"])
+    lib_re = re.compile(r'^(lib)?botan-3\.(so|a|dll|dylib|lib)$')
```

Comment:
I know the hard coded major version changes only once every few years: But wouldn't it be worthwhile to pull it from the `build_config['version_major']`? Or even better: Implement the `verify_library` similarly to the `verify_includes` where `./configure.py` stores a list of expected library files in the `build_config`?